src/gtc/passes/oir_optimizations/utils.py

```diff
@@ -160,8 +161,8 @@ def symbol_name_creator(used_names: Set[str]) -> Callable[[str], str]:
     """
 
     def increment_string_suffix(s: str) -> str:
-        if not s[-1].isnumeric():
-            return s + "0"
+        if not re.match(r".+_gen_[0-9]+", s):
```

Comment:
I would compile this regular expression at module level since it's constant. It won't probably make such a big difference but just in case

Reply:
Good idea. Implemented!